File: MDEV1004-M2023-Assignment3B/Server/Models/movie.ts
// File Name: movie.ts
// Student Name: Chesta Patel
// Student ID: 200542446
// Date: 22nd July 2023

import { Schema, model } from "mongoose";

interface IMovie {
    movieID: string;
    title: string;
    studio: string;
    genres: string[];
    directors: string[];
    writers: string[];
    actors: string[];
    year: number;
    length: number;
    shortDescription: string;
    mpaRating: string;
    criticsRating: number;
    posterLink: string;
}

let movieSchema = new Schema<IMovie>({
    movieID: String,
    title: String,
    studio: String,
    genres: [String],
    directors: [String],
    writers: [String],
    actors: [String],
    year: Number,
    length: Number,
    shortDescription: String,
    mpaRating: String,
    criticsRating: Number,
    posterLink: String,
});

let Movie = model<IMovie>("Movie", movieSchema);

export default Movie;

File: MDEV1004-M2023-Assignment3-iOS/MDEV1004-M2023-Assignment3-iOS/Movie.swift
//
//  Movie.swift
//  MDEV1004-M2023-Assignment3-iOS
//
//  Created by Chesta Patel on 22/07/23.
//

struct Movie: Codable
{
    let movieID: String
    let title: String
    let studio: String
    let genres: [String]
    let directors: [String]
    let writers: [String]
    let actors: [String]
    let year: Int
    let length: Int
    let shortDescription: String
    let mpaRating: String
    let criticsRating: Double
}
